hw/ip/pwm/data/pwm.hjson

```diff
@@ -0,0 +1,253 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+{ name: "pwm",
+  clock_primary: "clk_i",
+  bus_device: "tlul",
+  bus_host: "none",
+  regwidth: "32",
+  registers: [
+    { name: "CFG",
+      desc: "Configuration register",
+      swaccess: "rw",
+      fields: [
+        { bits: "31",
+          name: "CNTR_EN",
+          desc: '''Write 1 enables the PWM counter
+                   Writing 0 disables the PWM counter, also resets the counter
+                '''   
+          resval: "0x0"             
+        }
+        { bits: "8:0",
+          name: "PWM_EN",
+          desc: '''Individual PWM output enable signals
+                   Write 1 enables output.
+                   Write 0 disable the output.
+                '''
+          resval: "0x000"
+        }
+      ]
+    },
+    { name: "PERIOD",
+      desc: "Sets PWM period, enable counter",
+      swaccess: "rw",
+      fields: [
+        { bits: "31:0",
+          name: "PERIOD",
+          desc: "Sets the period"
+          resval: "0x00000090"
+        }
+      ]
+    },
+    { name: "POSEDGE_VAL_0",
```

Comment:
this is a minor point, but you could wrap these up in a multi-reg, which would significantly reduce the amount of typing. 

Both `alert_handler` and `flash_ctrl` have examples of this. I can point you to it if there's interest later for the main branch.

Reply:
I dislike changing things that simulate/synthesize just for stylistic issues.  We can discuss later.

Goal right now is getting something for bronze gate count and file exercise.

Reply:
it's up to you honestly. There's no requirement that you use something like `multireg`. I just was pointing it out there is such a convenience feature, but it is not a mandate.